pypeit/reduce.py

```diff
@@ -635,6 +644,18 @@ def global_skysub(self, skymask=None, update_crmask=True, trim_edg=(3,3),
         # Mask objects using the skymask? If skymask has been set by objfinding, and masking is requested, then do so
         skymask_now = skymask if (skymask is not None) else np.ones_like(self.sciImg.image, dtype=bool)
 
+        # Allow for previous sky to better estimate ivar
+        #  Unless we used a background image (i.e. ir_redux=True)
+        if previous_sky is None or self.ir_redux:
+            ivar = self.sciImg.ivar 
+        else:
+            # Estimate the variance using the input sky model
+            var = procimg.variance_model(self.sciImg.base_var, 
+                                          counts=previous_sky, 
+                                          count_scale=self.sciImg.img_scale, 
+                                          noise_floor=self.sciImg.noise_floor)
+            ivar = utils.inverse(var)
+
```

Comment:
@profxj I found that this fix creates an over subtraction in the spectrum with optimal extraction. This is due to the fact that when the optimal extraction is performed, the "new" global sky is used, but the old `self.sciImg.ivar` is used. I found that just updating here `self.sciImg.ivar` (instead of using `ivar`), takes care of it. I can make this change in the branch I am working on, or you can do it here. As you wish. I just wanted to let you know.

Reply:
@deborape -- Please add it to this branch.
Thx!

Reply:
Done!